compiler/rustc_expand/src/base.rs

```diff
@@ -1132,37 +1132,47 @@ impl<'a> ExtCtxt<'a> {
     }
 }
 
-/// Extracts a string literal from the macro expanded version of `expr`,
-/// emitting `err_msg` if `expr` is not a string literal. This does not stop
-/// compilation on error, merely emits a non-fatal error and returns `None`.
```

Comment:
Can you correct the comment instead of removing it?
For instance: what is the extra bool in the return type?

Reply:
I didn't remove the comment, I moved it down to `expr_to_string`, which I think it was meant for. It actually emits an error and returns an option as the comment says.. should I document `expr_to_spanned_string` as well?

Reply:
Yes, please.